File: app/src/main/java/com/stoffe/gym/Database/WorkoutViewModel.java
package com.stoffe.gym.Database;

import android.app.Application;

import java.util.List;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

public class WorkoutViewModel extends AndroidViewModel {

    WorkoutRepository workoutRepository;
    ExerciseRepository exerciseRepository;
    LiveData<List<Workout>> workoutList;
    private final MutableLiveData<Workout> currentWorkout;
    private final MutableLiveData<Exercise> currentExercise;

    public WorkoutViewModel(Application application){
        super(application);
        workoutRepository = new WorkoutRepository(application);
        exerciseRepository = new ExerciseRepository(application);
        workoutList = workoutRepository.getWorkout();
        currentWorkout = new MutableLiveData<>();
        currentExercise = new MutableLiveData<>();
    }

    public LiveData<List<Workout>> getAllWorkouts(){
        return workoutList;
    }

    public void insertWorkout(Workout workout){
        workoutRepository.insert(workout);
    }

    public void setCurrentWorkout(Workout workout){
        currentWorkout.setValue(workout);
    }

    public LiveData<Workout> getCurrentWorkout(){
        return currentWorkout;
    }

    public void setCurrentExercise(Exercise exercise){
        currentExercise.setValue(exercise);
    }

    public LiveData<Exercise> getCurrentExercise(){
        return currentExercise;
    }

    public void deleteWorkout(Workout workout){
        workoutRepository.delete(workout);
    }

    public LiveData<List<Exercise>> getAllExercises(){
        return exerciseRepository.getExercise();
    }

    public LiveData<List<Exercise>> getAllExercisesWithId(int id){
        return exerciseRepository.getExerciseWithID(id);
    }

    public void insertExercise(Exercise exercise){
        exerciseRepository.insertExercise(exercise);
    }

    public void deleteExercise(Exercise exercise){
        exerciseRepository.deleteExercise(exercise);
    }

    public LiveData<List<ExerciseData>> getAllExercisesDataWithId(int id){
        return exerciseRepository.getExerciseDataWithID(id);
    }

    public void insertExerciseData(ExerciseData exerciseData){
        exerciseRepository.insertExerciseData(exerciseData);
    }

    public void deleteExerciseData(ExerciseData exerciseData){
        exerciseRepository.deleteExerciseData(exerciseData);
    }
}

File: app/src/main/java/com/stoffe/gym/database/WorkoutViewModel.kt
package com.stoffe.gym.database

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.viewModelScope
import com.stoffe.gym.database.entities.Exercise
import com.stoffe.gym.database.entities.ExerciseData
import com.stoffe.gym.database.entities.Workout
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class WorkoutViewModel(application: Application) : AndroidViewModel(
    application
) {

    private val workoutRepository = WorkoutRepository(getDatabase(application))
    private val exerciseRepository = ExerciseRepository(getDatabase(application))
   // val summaryRepository: SummaryRepository
    val allWorkouts = workoutRepository.workouts
    private val currentWorkout: MutableLiveData<Workout>
    val currentExercise: MutableLiveData<Exercise>

    val allExercisesDataWithId = MutableStateFlow<List<ExerciseData>?>(null)

    fun getExercisesById(id: Int): Flow<List<Exercise>> {
        return exerciseRepository.getExerciseWithIDTest(id)
    }

    fun getExercisesDataById(id: Int): Flow<List<ExerciseData>> {
        return exerciseRepository.getExerciseDataWithID(id)
    }

    init {
       // summaryRepository = SummaryRepository(application)
        currentWorkout = MutableLiveData()
        currentExercise = MutableLiveData()
    }

    fun insertWorkout(workout: Workout) {
        viewModelScope.launch {
            workoutRepository.insert(workout)
        }

    }

    fun updateWorkout(workout: Workout) {
        viewModelScope.launch {
            workout.setActive(!workout.isActive)
            workoutRepository.update(workout)
        }
    }

    fun setCurrentWorkout(workout: Workout) {
        currentWorkout.value = workout
    }

    fun getCurrentWorkout(): LiveData<Workout> {
        return currentWorkout
    }

    fun setCurrentExercise(exercise: Exercise) {
        currentExercise.value = exercise
    }

    fun getCurrentExercise(): LiveData<Exercise> {
        return currentExercise
    }

    fun deleteWorkout(workout: Workout) {
        viewModelScope.launch {
            workoutRepository.delete(workout)
        }
    }

    fun insertExercise(exercise: Exercise) {
        viewModelScope.launch {
            exerciseRepository.insertExercise(exercise)
        }
    }

    fun deleteExercise(exercise: Exercise) {
        viewModelScope.launch {
            exerciseRepository.deleteExercise(exercise)
        }
    }

    fun insertExerciseData(exerciseData: ExerciseData) {
        viewModelScope.launch {
            exerciseRepository.insertExerciseData(exerciseData)
        }
    }

    fun deleteExerciseData(exerciseData: ExerciseData) {
        viewModelScope.launch {
            exerciseRepository.deleteExerciseData(exerciseData)
        }
    }

    /*
    val allSummaries: LiveData<List<Summary>>
        get() = summaryRepository.summary

     */

}